runtime/expr/ztests/cast-type.yaml

```diff
@@ -0,0 +1,13 @@
+zed: 'yield type(this)'
+
+input: |
+  "<int64>"
+  "<{x:string}>"
+  "1"
+  1.
+
+output: |
+  <int64>
+  <{x:string}>
+  error({message:"cannot cast to type",on:"1"})
+  error({message:"cannot cast to type",on:1.})
```

Comment:
I actually think it's toss up between whether there are more cast tests in `expr_test` or in the `ztests` directory. Personally I think its easier to find tests if they're ztests rather than buried in go tests. I kind of rather see that we move the tests in `expr_test` into ztests.

Reply:
I agree.  Much prefer ztests and remember fighting expr_test when we made changes.  At some point we should have a better way to ztest lots of different combos of things in one file.